packages/salesforcedx-vscode-soql/src/index.ts

```diff
@@ -5,21 +5,21 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
+import { WorkspaceContextUtil } from '@salesforce/salesforcedx-utils-vscode/out/src/context';
 import * as vscode from 'vscode';
 import { startLanguageClient, stopLanguageClient } from './client/client';
 import { soqlOpenNew } from './commands';
 import { SOQLEditorProvider } from './editor/soqlEditorProvider';
 import { QueryDataViewService } from './queryDataView/queryDataViewService';
 import { startTelemetry, stopTelemetry } from './telemetry';
-import { checkDependencies } from './sfdx';
 
 export async function activate(
   context: vscode.ExtensionContext
 ): Promise<void> {
   const extensionHRStart = process.hrtime();
-  checkDependencies();
   context.subscriptions.push(SOQLEditorProvider.register(context));
   QueryDataViewService.register(context);
+  WorkspaceContextUtil.getInstance().initialize(context);
```

Comment:
I think we should add an `await` here. If we don't then we run the risk of initializing the language server with an empty connection.

Reply:
Good idea. I made the change in `feat(soql): await workspace context in activation`.